### PR TITLE
zoxide: remove imports

### PR DESCRIPTION
**Pull request by ajeetdsouza**
###### Motivation for this change
zoxide no longer requires external programs in tests by default, so I've removed the extra imports.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/zoxide/default.nix
Comment (cole-h) [inline pkgs/tools/misc/zoxide/default.nix:24]: `doCheck` defaults to `true` :)

```suggestion
```

Comment (ajeetdsouza) [reply pkgs/tools/misc/zoxide/default.nix:24]: @cole-h thanks, fixed! Btw, do you know why `libiconv` is needed on macOS?

Comment (cole-h) [reply pkgs/tools/misc/zoxide/default.nix]: No clue, I don't use macOS.

**Review by cole-h — APPROVED** [review]
If you would squash all these commits into something like `zoxide: remove shell checkInputs`, that would be great :)

Just need someone to confirm it still builds (and tests succeed) on Darwin before merging. (cc @SuperSandro2000)

**Comment by ajeetdsouza** [comment]
@SuperSandro2000, it would be great if you could test removing `libiconv` too, I'm not sure why that dependency is required.

**Comment by happysalada** [comment]
@ajeetdsouza a couple of weeks ago when I tried compiling on macos it failed because of missing libiconv.
https://github.com/NixOS/nixpkgs/pull/118875
I added it in this PR.

**Comment by happysalada** [comment]
Just as added information, I had to add libiconv to a bunch of rust tools I use on mac. It might be some flag having been added in the cargo compilation.

**Comment by ysndr** [comment]
I experience the same, I think it's related to clap but I haven't tracked it down completely

**Comment by happysalada** [comment]
Result of `nixpkgs-review pr 119453` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>zoxide</li>
  </ul>
</details>

**Review by ysndr — APPROVED** [review]
Builds and runs on my system (MacOS 10.14)

**Comment by happysalada** [comment]
LGTM too

**Review by happysalada — APPROVED** [review]

**Comment by ajeetdsouza** [comment]
I see, thanks for clarifying! I think this should be ready to merge, then.

As a side note, I just wanted to thank you all for helping maintain zoxide! I'd add myself to the maintainer list too, but I don't know enough about Nix to provide useful feedback :sweat_smile: while I do use a `shell.nix` file for zoxide's CI, too, it's quite hacky due to my limited knowledge.

Someday I'll convert it to a Nix flake. Someday.

**Comment by happysalada** [comment]
I would say don't beat yourself over making a flake. I think it could be useful in case people want to run a specific commit, but I don't see it ad bringing a ton of value. I think working on your software is where the value is!

**Comment by ajeetdsouza** [comment]
Haha, thank you!

**Pull request merged**
